cmd/collector/app/handler/thrift_span_handler.go

```diff
@@ -131,6 +137,10 @@ func (h *zipkinSpanHandler) SubmitZipkinBatch(spans []*zipkincore.Span, options
 	return responses, nil
 }
 
+func (h *zipkinSpanHandler) Close() error {
+	return h.modelProcessor.Close()
```

Comment:
This looks like we might be closing this twice? It's the ownership issue again - handlers don't create processors, so they should not be closing them.

Reply:
You are absolutely right. This wasn't a problem because the processor's `#Close()` was checking whether it had been closed before. I kept this safety check, but extracted the building of the span processor into its own function. The collector now builds it explicitly and passes to the handler builder.

While I don't quite like that the collector calls `builder.BuildProcessor()` and then `builder.BuildHandlers(processor)`, I think it's the best option that makes sense without building another abstraction with its own `#Close()`.

Reply:
if Collector builds the processor explicitly, shouldn't it be in charge of closing it as well, rather than having the handlers do it?

Reply:
...

What if I told you that, after all that I wrote before, I say that I actually forgot to remove it from the handler's `#Close()` methods...

Hopefully, it's now fixed in this last commit.